M2/Macaulay2/e/hash.hpp

```diff
@@ -29,14 +30,15 @@ class EngineObject : public our_new_delete
   virtual unsigned int computeHashValue() const = 0;
 };
 
-class MutableEngineObject : public our_new_delete
+class MutableEngineObject : public our_gc_cleanup
```

Comment:
What is the difference between these two classes?

Reply:
@mikestillman -- this is for you

Reply:
`our_new_delete` is basically `gc` (a class in c++ interface of GC: new of an object of that class uses GC to allocate memory), whereas `our_gc_cleanup` matches `gc_cleanup`, which does the above, and finalizes the object.  We are using the former as we have used it for years, before the `gc` class was available in GC.  `gc_cleanup` I have had trouble getting to work, in debug mode.

I do see one issue here: for some computations (e.g. GB computations), we are explicitly calling finalizers (see `finalize.hpp`), and also doing it here.  I will look into this today.  @mikestillman